### PR TITLE
Don't restore corrupted sizes/positions when restoring layout items 

### PR DESCRIPTION
**Pull request by nyalldawson**
This can happen when eg a broken item size causes a nan position,
which breaks the layout size calculation and results in nan or
massive x/y values. Restoring these leads to a broken
layout which cannot be interacted with.

Refs https://github.com/qgis/QGIS/issues/55240

### CODE REVIEW

File: src/core/layout/qgslayoutpoint.cpp [hunk 1]
```diff
@@ -26,7 +26,12 @@ QgsLayoutPoint::QgsLayoutPoint( const double x, const double y, const Qgis::Layo
   , mY( y )
   , mUnits( units )
 {
-
+#ifdef QGISDEBUG
+  if ( std::isnan( mX ) || std::isnan( mY ) )
+  {
+    qWarning( "Layout point with NaN coordinates created" );
```
Comment (nirvn) [inline src/core/layout/qgslayoutpoint.cpp:32]: We don't want to Q_ASSERT on these scenarios to make it extra visible to devs?

**Review by nirvn — APPROVED** [review]

File: src/core/layout/qgslayoutpoint.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/core/layout/qgslayoutpoint.cpp]: Ok, done!

**Comment by nyalldawson** [comment]
@nirvn the assert triggered a few more bugs -- can you re-review please?

**Review by nirvn — APPROVED** [review]
We :heart:  asserts.

**Pull request merged**